design/one-pager-resource-status.md

```diff
@@ -0,0 +1,133 @@
+# Resource High Level Stop-Light Status
+
+- Owner: Steven Rathbauer ([@rathpc](https://github.com/rathpc))
+- Reviewers: Crossplane Maintainers
+- Status: Draft
+
+## Problem
+
+As a stack author there is currently no way to author rules that combine arbitrary conditions of a given state to
+represent high level _"stop-light"_ conditions.
+
+The high level conditions would be:
+- Online
+- Offline
+- Warning
+- Unknown
+
+The **Unknown** condition is interesting because this would be returned if the conditions were annotated but we cannot
+verify a known state based on the given rules. _(Essentially a default case in a switch block)_.
+
+## Design
+
+In contrast to the design [here](https://github.com/crossplaneio/crossplane/blob/master/design/one-pager-stack-status-spec.md),
+this design is meant to handle combinations of arbitrary statuses to determine a high level "overall" status.
+
+The following examples are meant to be added to a `ui-schema.yaml` file. The `resourceStatus` key should be added as a root key in that file.
+
+<a name="example-1"></a>
+
+```yaml
+# Example 1
+resourceStatus:
+  paths: .status.conditionedStatus.conditions
+  states:
+    Online:
+    - Ready: 'True'
+      Synchronized: 'True'
+    Offline:
+    - Ready: 'False'
+      Synchronized: 'False'
+    Warning:
+    - Ready: 'False'
+      Synchronized: 'True'
+      SyncFailure: 'True'
+```
```

Comment:
What's the thinking behind defining separate matching logic for the offline and online states? Are there cases in which offline can't be inferred by online being false?

Also, do we want to align the terms used here with those that Crossplane is using? Most (all?) of our resources currently use "available and unavailable" rather than "online and offline" - https://godoc.org/github.com/crossplaneio/crossplane-runtime/apis/core/v1alpha1#Condition

Reply:
The idea is that the logic behind each case is entirely arbitrary and up to the stack author to decide. This is also why there are technically 4 states (Online, Offline, Warning and Unknown).

So technically there can be a case where offline is not specifically the direct negation of online and warning or unknown can also play a part in anything in between.